### PR TITLE
CASSANDRA-19900: Make the compression cache configurable to reduce he…

### PR DESCRIPTION
**Pull request by jberragan**
…ap pressure for large SSTables

### CODE REVIEW

File: cassandra-analytics-common/src/main/java/org/apache/cassandra/spark/utils/Properties.java [hunk 1]
```diff
@@ -40,6 +40,8 @@ public final class Properties
     public static final int DEFAULT_MAX_RETRIES = 10;
     public static final long DEFAULT_MILLIS_TO_SLEEP = 500;
     public static final int DEFAULT_MAX_POOL_SIZE = 64;
+    public static final boolean DEFAULT_CACHE_COMPRESSION_METADATA_KEY = true;
+    public static final int DEFAULT_MAX_SIZE_CACHE_COMPRESSION_METADATA_KEY = 8388608; // 8MiB
```
Comment (frankgh) [inline cassandra-analytics-common/src/main/java/org/apache/cassandra/spark/utils/Properties.java]:
```suggestion
    public static final int DEFAULT_MAX_SIZE_CACHE_COMPRESSION_METADATA_KEY = 8 * MEBI_BYTES; // 8MiB
```

File: cassandra-analytics-core/src/main/java/org/apache/cassandra/clients/Sidecar.java [hunk 2]
```diff
@@ -210,6 +212,8 @@ public static final class ClientConfig
         public static final String CHUNK_BUFFER_SIZE_BYTES_KEY = "chunkBufferSizeBytes";
         public static final String MAX_POOL_SIZE_KEY = "maxPoolSize";
         public static final String TIMEOUT_SECONDS_KEY = "timeoutSeconds";
+        public static final String CACHE_COMPRESSION_METADATA_KEY = "cacheCompressionMetadata";
+        public static final String MAX_SIZE_CACHE_COMPRESSION_METADATA_KEY = "maxCacheCompressionMetadata";
```
Comment (frankgh) [inline cassandra-analytics-core/src/main/java/org/apache/cassandra/clients/Sidecar.java]: Can we add the units for this param?
```suggestion
        public static final String MAX_SIZE_CACHE_COMPRESSION_METADATA_KEY = "maxCacheCompressionMetadataMiB";
```

File: cassandra-analytics-core/src/main/java/org/apache/cassandra/clients/Sidecar.java [hunk 3]
```diff
@@ -398,7 +414,19 @@ public static ClientConfig create(int userProvidedPort,
                                     maxPoolSize,
                                     timeoutSeconds,
                                     maxBufferOverride,
-                                    chunkBufferOverride);
+                                    chunkBufferOverride,
+                                    cacheCompressionMetadata,
+                                    maxSizeCacheCompressionMetadata);
+        }
+
+        public int maxSizeCacheCompressionMetadata()
```
Comment (frankgh) [inline cassandra-analytics-core/src/main/java/org/apache/cassandra/clients/Sidecar.java]: can we add the units here as well?

**Review by frankgh — COMMENTED** [review]
Added some small comments

**Comment by jberragan** [comment]
Green CI: https://app.circleci.com/pipelines/github/jberragan/cassandra-analytics/81/workflows/b2fc5c83-7fb3-49a9-b9d1-63ed894d08b0

**Review by frankgh — APPROVED** [review]
+1 Looks good to me

File: cassandra-analytics-common/src/main/java/org/apache/cassandra/spark/utils/Properties.java (hunk 1 above)
Comment (belliottsmith) [reply cassandra-analytics-common/src/main/java/org/apache/cassandra/spark/utils/Properties.java]: for future reference, shifting left in multiples of 10 multiplies by KiB. so 8 << 20 is a clean way to say 8 Mebibytes

**Pull request merged**